### PR TITLE
docs: add jsdoc comments

### PR DESCRIPTION
**Pull request by andreialecu**
**What's the problem this PR addresses?**

Adds some jsdoc comments

**How did you fix it?**

With my keyboard.

**Checklist**

- [x] I have read the [Contributing Guide](https://yarnpkg.com/advanced/contributing).

- [x] I have set the packages that need to be released for my changes to be effective.

- [x] I will check that all automated PR checks pass before the PR gets reviewed.

### CODE REVIEW

File: packages/yarnpkg-core/sources/structUtils.ts [hunk 1]
```diff
@@ -14,37 +14,73 @@ import {Ident, Descriptor, Locator, Package}    from './types';
...
+ */
 export function makeIdent(scope: string | null, name: string): Ident {
   if (scope?.startsWith(`@`))
     throw new Error(`Invalid scope: don't prefix it with '@'`);
 
   return {identHash: hashUtils.makeHash<IdentHash>(scope, name), scope, name};
 }
 
+/**
+ * Creates an descriptor for a package
```
Comment (merceyz) [inline packages/yarnpkg-core/sources/structUtils.ts:30]:
```suggestion
 * Creates a descriptor for a package
```

File: packages/yarnpkg-core/sources/structUtils.ts [hunk 2]
```diff
@@ -14,37 +14,73 @@ import {Ident, Descriptor, Locator, Package}    from './types';
...
+ * @param ident the ident of the package (see `makeIdent`)
+ * @param range the range (for example `^1.0.0`)
+ */
 export function makeDescriptor(ident: Ident, range: string): Descriptor {
   return {identHash: ident.identHash, scope: ident.scope, name: ident.name, descriptorHash: hashUtils.makeHash<DescriptorHash>(ident.identHash, range), range};
 }
 
+/**
+ * Creates an locator for a package
```
Comment (merceyz) [inline packages/yarnpkg-core/sources/structUtils.ts:39]:
```suggestion
 * Creates a locator for a package
```

File: packages/yarnpkg-core/sources/structUtils.ts [hunk 3]
```diff
@@ -319,6 +473,7 @@ export function parseFileStyleRange(range: string, {protocol}: {protocol: string
   const parentLocator = parseLocator(params.locator, true);
   const path = selector as PortablePath;
 
+  console.log(arguments, {parentLocator, path})
```
Comment (merceyz) [inline packages/yarnpkg-core/sources/structUtils.ts:476]:
```suggestion
```

File: packages/yarnpkg-core/sources/structUtils.ts [hunk 4]
```diff
@@ -460,37 +659,76 @@ function prettyRangeNoColors(range: string): string {
   }
 }
 
+/**
+ * Returns a `string` that is suitable to be printed to `stdout` (contains colors)
```
Comment (merceyz) [inline packages/yarnpkg-core/sources/structUtils.ts:663]: > contains colors

Doesn't that depend if you have it enabled or not?

File: packages/yarnpkg-core/sources/types.ts [hunk 5]
```diff
@@ -2,46 +2,164 @@ import {PortablePath}                       from '@yarnpkg/fslib';
...
+   */
   scope: string | null,
+  /**
+   * Name of the package (eg. `node`)
+   */
   name: string,
 }
 
+/**
+ * Unique hash of a package scope, name and version range.
```
Comment (merceyz) [inline packages/yarnpkg-core/sources/types.ts:37]:
```suggestion
 * Unique hash of a package scope, name, and version range.
```

File: packages/yarnpkg-core/sources/structUtils.ts (hunk 4 above)
Comment (andreialecu) [reply packages/yarnpkg-core/sources/structUtils.ts:662]: It does. Rephrasing suggestions appreciated :)

**Comment by arcanis** [comment]
Made a few changes, thanks for looking into this!

**Pull request merged**